utils/helpers.py

```
@@ -0,0 +1,190 @@
+"""
+Helper utility functions for the self-correcting MCP runtime
+"""
+
+import json
+import time
+import hashlib
+from typing import Any, Dict, List, Optional, Union
+from pathlib import Path
+
+
+def safe_json_parse(json_string: str) -> Optional[Dict[str, Any]]:
```

Ruff (0.11.9)

Use `dict` instead of `Dict` for type annotation

Replace with `dict`

(UP006)